### PR TITLE
add mkdocs documentation

### LINKED ISSUE #435: create a dedicate documentation site
we should follow the same as demonstrated in https://github.com/aquasecurity/starboard , and use mkdocs, with similar design

### PR DESCRIPTION
**Pull request by ronaudinho**
resolves #435

the docs are taken from existing READMEs with some modifications, as far as I remember, tracee has different project structure a few months back, so there are a number of overlaps especially between the root README and tracee-ebpf/README, which is reflected in the current mkdocs structure as well.

### CODE REVIEW

File: .github/workflows/mkdocs-dev.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        run: |
+          pip install git+https://${GH_TOKEN}@github.com/squidfunk/mkdocs-material-insiders.git
+          pip install mike
+          pip install mkdocs-macros-plugin
+        env:
+          GH_TOKEN: ${{ secrets.ORG_GITHUB_TOKEN }}
+      - name: Configure the git user
+        run: |
+          git config user.name "ronaudinho"
+          git config user.email "doelaudi@gmail.com"
```
Comment (ronaudinho) [inline .github/workflows/mkdocs-dev.yaml:32]: might want to adjust this @itaysk

File: .github/workflows/mkdocs-latest.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        run: |
+          pip install git+https://${GH_TOKEN}@github.com/squidfunk/mkdocs-material-insiders.git
+          pip install mike
+          pip install mkdocs-macros-plugin
+        env:
+          GH_TOKEN: ${{ secrets.ORG_GITHUB_TOKEN }}
+      - name: Configure the git user
+        run: |
+          git config user.name "ronaudinho"
+          git config user.email "doelaudi@gmail.com"
```
Comment (ronaudinho) [inline .github/workflows/mkdocs-latest.yaml:29]: ditto for this

File: docs/rules/rego.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+Create a `.rego` file in the rules directory that has the following Rego Rules (in this context rules are Rego's language constructs):
+
+1. `__rego_metadoc__`: A *document* rule that defines the rule's metadata (based on [WIP Rego convention](https://hackmd.io/@ZtQnh19kS26YiNlJLqKJnw/H1gAv5nB) for describing policy metadata).
```
Comment (ronaudinho) [inline docs/rules/rego.md:3]: the link returns not found

Comment (itaysk) [reply docs/rules/rego.md:3]: you can remove whatever's in parenthesis

File: .github/workflows/mkdocs-dev.yaml (hunk 1 above)
Comment (itaysk) [reply .github/workflows/mkdocs-dev.yaml:32]: can we just remove these lines? in other workflows where we git push without overriding the username, it shows as the "Github Actions" user, which is better I think

File: .github/workflows/mkdocs-latest.yaml (hunk 2 above)
Comment (itaysk) [reply .github/workflows/mkdocs-latest.yaml:29]: see previous comment

File: mkdocs.yml [hunk 4]
```diff
@@ -0,0 +1,64 @@
+site_name: Tracee
+site_url: https://aquasecurity.github.io/tracee/
+site_description: Tracee is a runtime security and forensics tool for Linux, using Linux eBPF technology to trace your system and applications at runtime, and analyze collected events to detect suspicious behavioral patterns.
```
Comment (itaysk) [inline mkdocs.yml:3]:
```suggestion
site_description: Runtime security and forensics using eBPF.
```

**Review by itaysk — COMMENTED** [review]
good job! I think I'll follow up with another PR that re-arranges the docs, but that's great as a starting point.
let's address the minor comments and then we can merge

**Comment by grantseltzer** [comment]
Would this then allow us to turn on github pages and have a webpage with the documentation?

**Comment by itaysk** [comment]
> Would this then allow us to turn on github pages and have a webpage with the documentation?

yes that's the goal

**Review by itaysk — APPROVED** [review]

**Pull request merged**